Error while checking build/O2/fullCI for 26a1ceae379cedbf98f81b16be5835fdf508bcd1 at 2023-10-26 11:00:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.


## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile reco_NOGPU.log
reco_NOGPU.log:[49674:calib-emcalchannel-badchannel]: [09:00:01][FATAL] error while setting up workflow in o2-calibration-emcal-channel-calib-workflow: filesystem error: cannot create directories: Permission denied [/scratch/services/detector_tmp/emc_calib]
[49670:BadMapCalibSpec]: [08:59:53][ERROR] Insufficient statistics: 10 entries in lowE histo, do nothing
[49674:calib-emcalchannel-badchannel]: [09:00:01][FATAL] error while setting up workflow in o2-calibration-emcal-channel-calib-workflow: filesystem error: cannot create directories: Permission denied [/scratch/services/detector_tmp/emc_calib]
[ERROR] pid 49674 (calib-emcalchannel-badchannel) crashed with or was killed with exit code 1
[ERROR]  - Device calib-emcalchannel-badchannel: pid 49674 (exit 1)
[INFO]    - First error: [09:00:01][FATAL] error while setting up workflow in o2-calibration-emcal-channel-calib-workflow: filesystem error: cannot create directories: Permission denied [/scratch/services/detector_tmp/emc_calib]
[ERROR] SEVERE: Device calib-emcalchannel-badchannel (49674) had at least one message above severity 5: error while setting up workflow in o2-calibration-emcal-channel-calib-workflow: filesystem error: cannot create directories: Permission denied [/scratch/services/detector_tmp/emc_calib]


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12069/26a1ceae379cedbf98f81b16be5835fdf508bcd1/build_O2_fullCI/pretty.html).

--- a/Detectors/TRD/qc/src/Tracking.cxx
+++ b/Detectors/TRD/qc/src/Tracking.cxx
@@ -30,18 +30,12 @@ void Tracking::init()
 
 void Tracking::setInput(const o2::globaltracking::RecoContainer& input)
 {
-  // mRecoCont = &input;
   mTracksTPC = input.getTPCTracks();
   mTracksITSTPC = input.getTPCITSTracks();
   mTracksITSTPCTRD = input.getITSTPCTRDTracks<TrackTRD>();
   mTracksTPCTRD = input.getTPCTRDTracks<TrackTRD>();
   mTrackletsRaw = input.getTRDTracklets();
   mTrackletsCalib = input.getTRDCalibratedTracklets();
-  if (mTracksTPC.size() > 0) {
-    LOG(info) << "Checking first TPC track";
-    auto trk = mTracksTPC[0];
-    LOG(info) << "Track has X of " << trk.getX();
-  }
 }
 
 void Tracking::run()

--- a/Detectors/TRD/reconstruction/src/CruRawReader.cxx
+++ b/Detectors/TRD/reconstruction/src/CruRawReader.cxx
@@ -494,8 +494,8 @@ bool CruRawReader::processHalfCRU(int iteration)
     int halfCruIdx = cruIdx * 2 + mFEEID.endpoint;                        // endpoint (0 or 1) defines half-CRU
     int linkIdxGlobal = halfCruIdx * NLINKSPERHALFCRU + currentlinkindex; // global link ID [0..1079]
     int halfChamberId = mLinkMap->getHCID(linkIdxGlobal);
-    mEventRecords.getCurrentEventRecord().getCounters().mLinkWords[halfChamberId] = mCurrentHalfCRULinkLengths[currentlinkindex];
-    mEventRecords.getCurrentEventRecord().getCounters().mLinkErrorFlag[halfChamberId] = mCurrentHalfCRULinkErrorFlags[currentlinkindex];
+    mEventRecords.getCurrentEventRecord().getCounters().mLinkWords[halfChamberId] += mCurrentHalfCRULinkLengths[currentlinkindex];
+    mEventRecords.getCurrentEventRecord().getCounters().mLinkErrorFlag[halfChamberId] |= mCurrentHalfCRULinkErrorFlags[currentlinkindex];
     mEventRecords.incLinkErrorFlags(halfChamberId, mCurrentHalfCRULinkErrorFlags[currentlinkindex]); // TODO maybe has more meaning on a per event basis?
     mEventRecords.incLinkWords(halfChamberId, mCurrentHalfCRULinkLengths[currentlinkindex]);
     uint32_t currentlinksize32 = mCurrentHalfCRULinkLengths[currentlinkindex] * 8; // x8 to go from 256 bits to 32 bit;

--- a/Detectors/TRD/workflow/io/include/TRDWorkflowIO/TRDDigitReaderSpec.h
+++ b/Detectors/TRD/workflow/io/include/TRDWorkflowIO/TRDDigitReaderSpec.h
@@ -32,7 +32,7 @@ namespace trd
 class TRDDigitReaderSpec : public o2::framework::Task
 {
  public:
-  TRDDigitReaderSpec(bool useMC) : mUseMC(useMC) {}
+  TRDDigitReaderSpec(bool useMC, bool useTriggerRecords, int subSpec) : mUseMC(useMC), mUseTriggerRecords(useTriggerRecords), mSubSpec(subSpec) {}
   ~TRDDigitReaderSpec() override = default;
   void init(o2::framework::InitContext& ic) override;
   void run(o2::framework::ProcessingContext& pc) override;
@@ -40,6 +40,8 @@ class TRDDigitReaderSpec : public o2::framework::Task
  private:
   void connectTree();
   bool mUseMC = false;
+  bool mUseTriggerRecords = true;
+  unsigned int mSubSpec = 1;
   std::unique_ptr<TFile> mFile;
   std::unique_ptr<TTree> mTreeDigits;
   std::string mFileName = "trddigits.root";
@@ -52,7 +54,7 @@ class TRDDigitReaderSpec : public o2::framework::Task
   o2::dataformats::IOMCTruthContainerView* mLabels = nullptr;
 };
 
-o2::framework::DataProcessorSpec getTRDDigitReaderSpec(bool useMC);
+o2::framework::DataProcessorSpec getTRDDigitReaderSpec(bool useMC, bool trigRec = true, int dataSubspec = 1);
 
 } // end namespace trd
 } // end namespace o2

--- a/Detectors/TRD/workflow/io/src/TRDDigitReaderSpec.cxx
+++ b/Detectors/TRD/workflow/io/src/TRDDigitReaderSpec.cxx
@@ -43,7 +43,9 @@ void TRDDigitReaderSpec::connectTree()
   mTreeDigits.reset((TTree*)mFile->Get(mDigitTreeName.c_str()));
   assert(mTreeDigits);
   mTreeDigits->SetBranchAddress(mDigitBranchName.c_str(), &mDigitsPtr);
-  mTreeDigits->SetBranchAddress(mTriggerRecordBranchName.c_str(), &mTriggerRecordsPtr);
+  if (mUseTriggerRecords) {
+    mTreeDigits->SetBranchAddress(mTriggerRecordBranchName.c_str(), &mTriggerRecordsPtr);
+  }
   if (mUseMC) {
     mTreeDigits->SetBranchAddress(mMCLabelsBranchName.c_str(), &mLabels);
   }
@@ -55,10 +57,12 @@ void TRDDigitReaderSpec::run(ProcessingContext& pc)
   auto currEntry = mTreeDigits->GetReadEntry() + 1;
   assert(currEntry < mTreeDigits->GetEntries()); // this should not happen
   mTreeDigits->GetEntry(currEntry);
-  LOG(info) << "Pushing " << mTriggerRecords.size() << " TRD trigger records at entry " << currEntry;
-  pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "TRKTRGRD", 1, Lifetime::Timeframe}, mTriggerRecords);
-  LOG(info) << "Pushing " << mDigits.size() << " digits for these trigger records";
-  pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "DIGITS", 1, Lifetime::Timeframe}, mDigits);
+  LOGP(info, "Pushing {} digits for tree entry {}", mDigits.size(), currEntry);
+  pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "DIGITS", mSubSpec, Lifetime::Timeframe}, mDigits);
+  if (mUseTriggerRecords) {
+    LOGP(info, "Pushing {} trigger records for tree entry {}", mTriggerRecords.size(), currEntry);
+    pc.outputs().snapshot(Output{o2::header::gDataOriginTRD, "TRKTRGRD", mSubSpec, Lifetime::Timeframe}, mTriggerRecords);
+  }
   if (mUseMC) {
     auto& sharedlabels = pc.outputs().make<o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>>(Output{o2::header::gDataOriginTRD, "LABELS", 0, Lifetime::Timeframe});
     mLabels->copyandflatten(sharedlabels);
@@ -69,18 +73,20 @@ void TRDDigitReaderSpec::run(ProcessingContext& pc)
   }
 }
 
-DataProcessorSpec getTRDDigitReaderSpec(bool useMC)
+DataProcessorSpec getTRDDigitReaderSpec(bool useMC, bool trigRec, int dataSubspec)
 {
   std::vector<OutputSpec> outputs;
-  outputs.emplace_back("TRD", "DIGITS", 1, Lifetime::Timeframe);
-  outputs.emplace_back("TRD", "TRKTRGRD", 1, Lifetime::Timeframe);
+  outputs.emplace_back("TRD", "DIGITS", dataSubspec, Lifetime::Timeframe);
+  if (trigRec) {
+    outputs.emplace_back("TRD", "TRKTRGRD", dataSubspec, Lifetime::Timeframe);
+  }
   if (useMC) {
     outputs.emplace_back("TRD", "LABELS", 0, Lifetime::Timeframe);
   }
   return DataProcessorSpec{"TRDDIGITREADER",
                            Inputs{},
                            outputs,
-                           AlgorithmSpec{adaptFromTask<TRDDigitReaderSpec>(useMC)},
+                           AlgorithmSpec{adaptFromTask<TRDDigitReaderSpec>(useMC, trigRec, dataSubspec)},
                            Options{
                              {"digitsfile", VariantType::String, "trddigits.root", {"Input data file containing TRD digits"}},
                              {"input-dir", VariantType::String, "none", {"Input directory"}}}};

--- a/Detectors/TRD/workflow/io/src/digit-reader-workflow.cxx
+++ b/Detectors/TRD/workflow/io/src/digit-reader-workflow.cxx
@@ -23,7 +23,9 @@ void customize(std::vector<ConfigParamSpec>& workflowOptions)
 {
   // option allowing to set parameters
   std::vector<o2::framework::ConfigParamSpec> options{
-    {"disable-mc", o2::framework::VariantType::Bool, true, {"disable MC propagation"}},
+    {"disable-mc", VariantType::Bool, true, {"disable MC propagation"}},
+    {"disable-trigrec", VariantType::Bool, false, {"disable trigger record reading when these come from tracklet reader"}},
+    {"digit-subspec", VariantType::Int, 1, {"allow overwriting default output data subspec"}},
     {"configKeyValues", VariantType::String, "", {"Semicolon separated key=value strings ..."}}};
   o2::raw::HBFUtilsInitializer::addConfigOption(options);
   std::swap(workflowOptions, options);
@@ -39,7 +41,9 @@ WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
   o2::conf::ConfigurableParam::updateFromString(configcontext.options().get<std::string>("configKeyValues"));
 
   auto useMC = !configcontext.options().get<bool>("disable-mc");
+  auto sendTriggerRecords = !configcontext.options().get<bool>("disable-trigrec");
+  auto dataSubspec = configcontext.options().get<int>("digit-subspec");
   WorkflowSpec specs;
-  specs.emplace_back(o2::trd::getTRDDigitReaderSpec(useMC));
+  specs.emplace_back(o2::trd::getTRDDigitReaderSpec(useMC, sendTriggerRecords, dataSubspec));
   return specs;
 }

--- a/Detectors/TRD/workflow/src/TRDTrackletTransformerWorkflow.cxx
+++ b/Detectors/TRD/workflow/src/TRDTrackletTransformerWorkflow.cxx
@@ -35,6 +35,7 @@ void customize(std::vector<ConfigParamSpec>& workflowOptions)
 {
   std::vector<o2::framework::ConfigParamSpec> options{
     {"disable-mc", o2::framework::VariantType::Bool, false, {"Disable MC labels"}},
+    {"disable-irframe-reader", o2::framework::VariantType::Bool, false, {"Don't read ITS IR frames from file"}},
     {"disable-root-input", o2::framework::VariantType::Bool, false, {"disable root-files input reader"}},
     {"disable-root-output", o2::framework::VariantType::Bool, false, {"disable root-files output writer"}},
     {"filter-trigrec", o2::framework::VariantType::Bool, false, {"ignore interaction records without ITS data"}},
@@ -61,7 +62,7 @@ WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
   }
 
   auto trigRecFilterActive = configcontext.options().get<bool>("filter-trigrec");
-  if (trigRecFilterActive) {
+  if (trigRecFilterActive && !configcontext.options().get<bool>("disable-irframe-reader")) {
     o2::globaltracking::InputHelper::addInputSpecsIRFramesITS(configcontext, spec);
   }
 

--- a/prodtests/full-system-test/dpl-workflow.sh
+++ b/prodtests/full-system-test/dpl-workflow.sh
@@ -339,7 +339,7 @@ if [[ ! -z ${WORKFLOW_DETECTORS_USE_GLOBAL_READER} ]]; then
   has_detector MCH && has_detector_from_global_reader MCH && add_W o2-mch-digits-reader-workflow "$DISABLE_MC --hbfutils-config o2_tfidinfo.root --infile mchdigits.root"
   has_detector MCH && has_detector_from_global_reader MCH && add_W o2-mch-clusters-reader-workflow "--hbfutils-config o2_tfidinfo.root" "" 0
   has_detector MCH && has_detector_from_global_reader MCH && add_W o2-mch-preclusters-reader-workflow "--hbfutils-config o2_tfidinfo.root" "" 0
-  has_detector TRD && has_detector_from_global_reader TRD && add_W o2-trd-digit-reader-workflow "--hbfutils-config o2_tfidinfo.root"
+  has_detector TRD && has_detector_from_global_reader TRD && add_W o2-trd-digit-reader-workflow "$DISABLE_MC --digit-subspec 0 --disable-trigrec --hbfutils-config o2_tfidinfo.root"
   has_detector TOF && has_detector_from_global_reader TOF && add_W o2-tof-reco-workflow "$DISABLE_MC --input-type digits --output-type NONE --hbfutils-config o2_tfidinfo.root"
 fi
 
@@ -463,7 +463,7 @@ fi
 (has_detector_reco TOF || has_detector_ctf TOF) && ! has_detector_from_global_reader TOF && add_W o2-tof-reco-workflow "$TOF_CONFIG --input-type $TOF_INPUT --output-type $TOF_OUTPUT $DISABLE_DIGIT_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC --pipeline $(get_N tof-compressed-decoder TOF RAW 1),$(get_N TOFClusterer TOF REST 1)"
 has_detector_reco ITS && ! has_detector_from_global_reader ITS && add_W o2-its-reco-workflow "--trackerCA $ITS_CONFIG $DISABLE_MC $DISABLE_DIGIT_CLUSTER_INPUT $DISABLE_ROOT_OUTPUT --pipeline $(get_N its-tracker ITS REST 1 ITSTRK)" "$ITS_CONFIG_KEY;$ITSMFT_STROBES;$ITSEXTRAERR"
 has_detector_reco FT0 && ! has_detector_from_global_reader FT0 && add_W o2-ft0-reco-workflow "$DISABLE_DIGIT_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC --pipeline $(get_N ft0-reconstructor FT0 REST 1)"
-has_detector_reco TRD && ! has_detector_from_global_reader TRD && add_W o2-trd-tracklet-transformer "$DISABLE_DIGIT_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC $TRD_FILTER_CONFIG --pipeline $(get_N TRDTRACKLETTRANSFORMER TRD REST 1 TRDTRKTRANS)"
+has_detector_reco TRD && (has_detector_matching ITSTPCTRD || has_detector_matching TPCTRD) && ! has_detector_from_global_reader TRD && add_W o2-trd-tracklet-transformer "--disable-irframe-reader $DISABLE_DIGIT_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC $TRD_FILTER_CONFIG --pipeline $(get_N TRDTRACKLETTRANSFORMER TRD REST 1 TRDTRKTRANS)"
 has_detectors_reco ITS TPC && has_detector_matching ITSTPC && add_W o2-tpcits-match-workflow "$DISABLE_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC $SEND_ITSTPC_DTGL  $TPC_CORR_SCALING --nthreads $ITSTPC_THREADS --pipeline $(get_N itstpc-track-matcher MATCH REST $ITSTPC_THREADS TPCITS)" "$ITSTPC_CONFIG_KEY;$INTERACTION_TAG_CONFIG_KEY;$ITSMFT_STROBES;$ITSEXTRAERR"
 has_detector_reco TRD && [[ ! -z "$TRD_SOURCES" ]] && add_W o2-trd-global-tracking "$DISABLE_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC $TRD_CONFIG $TRD_FILTER_CONFIG $TPC_CORR_SCALING --track-sources $TRD_SOURCES --pipeline $(get_N trd-globaltracking_TPC_ITS-TPC_ TRD REST 1 TRDTRK),$(get_N trd-globaltracking_TPC_FT0_ITS-TPC_ TRD REST 1 TRDTRK),$(get_N trd-globaltracking_TPC_FT0_ITS-TPC_CTP_ TRD REST 1 TRDTRK)" "$TRD_CONFIG_KEY;$INTERACTION_TAG_CONFIG_KEY;$ITSMFT_STROBES;$ITSEXTRAERR"
 has_detector_reco TOF && [[ ! -z "$TOF_SOURCES" ]] && add_W o2-tof-matcher-workflow "$DISABLE_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC $TPC_CORR_SCALING --track-sources $TOF_SOURCES --pipeline $(get_N tof-matcher TOF REST 1 TOFMATCH)" "$ITSMFT_STROBES;$ITSEXTRAERR"
@@ -492,7 +492,7 @@ has_detectors_reco MFT MCH && has_detector_matching MFTMCH && add_W o2-globalfwd
 
 # ---------------------------------------------------------------------------------------------------------------------
 # Writers for detectors whose reco workflow cannot write the output
-has_detector_reco TRD && ( [[ -z "$DISABLE_ROOT_OUTPUT" ]] || needs_root_output o2-trd-digittracklet-writer ) && ! has_detector_from_global_reader TRD && add_W o2-trd-digittracklet-writer
+[[ $CTFINPUT == 1 ]] || [[ $DIGITINPUT == 1 ]] && has_detector_reco TRD && ( [[ -z "$DISABLE_ROOT_OUTPUT" ]] || needs_root_output o2-trd-digittracklet-writer ) && ! has_detector_from_global_reader TRD && add_W o2-trd-digittracklet-writer
 has_detector_reco TPC && [[ "0${TPC_CONVERT_LINKZS_TO_RAW:-}" == "01" ]] && ( [[ -z "$DISABLE_ROOT_OUTPUT" ]] || needs_root_output o2-gpu-reco-workflow ) && ! has_detector_from_global_reader TPC && add_W o2-tpc-reco-workflow "--input-type digitizer --output-type digits $DISABLE_MC"
 has_detector_reco CPV && ( [[ -z "$DISABLE_ROOT_OUTPUT" ]] || needs_root_output o2-cpv-cluster-writer-workflow ) && ! has_detector_from_global_reader CPV && add_W o2-cpv-cluster-writer-workflow "$DISABLE_MC"
 ( workflow_has_parameter AOD || [[ -z "$DISABLE_ROOT_OUTPUT" ]] || needs_root_output o2-emcal-cell-writer-workflow ) && ! has_detector_from_global_reader EMC && has_detector EMC && add_W o2-emcal-cell-writer-workflow "$DISABLE_MC --subspec 10 --cell-writer-name emcal-led-cells-writer --emcal-led-cells-writer \"--outfile emcledcells.root\""

